hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/config/internal/KeyValuePairConverterTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.config.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.hedera.node.app.spi.config.types.KeyValuePair;
+import com.swirlds.common.config.sources.SimpleConfigSource;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+class KeyValuePairConverterTest {
+
+    @Test
+    void testNullValue() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = null;
+
+        // then
+        assertThatThrownBy(() -> converter.convert(value)).isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void testValidValue() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key;value";
+
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value");
+    }
+
+    @Test
+    void testValidValueMultipleSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key;value;stillValue;";
+
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.key()).isEqualTo("key");
+        assertThat(result.value()).isEqualTo("value;stillValue;");
+    }
+
+    @Test
+    void testInvalidValueNoSemicolon() {
+        // given
+        final KeyValuePairConverter converter = new KeyValuePairConverter();
+        final String value = "key";
+
+        // then
+        assertThatThrownBy(() -> converter.convert(value)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testIncludedInConfig() {
+        // given
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new KeyValuePairConverter())
+                .withSource(new SimpleConfigSource("key", "key;value"))
+                .build();
+
+        // when
+        final KeyValuePair value = configuration.getValue("key", KeyValuePair.class);
```

Comment:
It was unexpected for me that this actually works.

Reply:
I assume you problem was with the naming. I did it more concrete.

Reply:
No, I was surprised, that you can get a single `KeyValuePair`. What happens, if more than one pair was specified? I wonder, if we really need this.

Reply:
I still do not get the question.

Reply:
The new functionality was added to allow the specification of a list of `KeyValuePair`s. I am surprised that you can call `configuration.getValue()` and get something back. I would have expected an exception.

What happens, if you call `configuration.getValue()` for a regular list of properties? What happens, if more than one element is defined?

Reply:
ah, no I understand.That is something that is not 100% perfect in the Config API and I would like to change in future. Currently a List is always defined as a comma separated string. If you want to get a `List` internally the raw string is always splitted (by `,`) and than all parts are converted individually. That is not a good solution (especially when YAML, JSON, ... comes into play) and hopefully will be fixed by a totally different issue in future.

**BUT:** For the comma based approach you can not say if a value is a collection with 1 element or a single element. Based on that it is handled the same for now (you can request a collect or a value). So technically the current solution is correct even if a single `KeyValuePair` does not make any sense.

Reply:
If I understand you correctly, if property is set to "a, b, c", you either get a list [a, b, c] by calling `getValues()` or a String "a, b, c" by calling `getValue()`. Is that correct?

In that case, I would also return a String "key;value" when `getValue()` is called in the example above. That would be consistent with the way how lists are handled IMO. Otherwise, what would you return, if the property is set to "key1;value1,key2;value2" and I call `getValue()`?

Reply:
> In that case, I would also return a String "key;value" when getValue() is called in the example above
That is exactly what happens if you do not define the return type when calling getValue(). If you define the type the config will always use the converter for the type if it has one.